core/src/main/java/org/apache/iceberg/rest/auth/HttpRequestFacade.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.auth;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A facade for an HTTP request that allows reading and modifying the request. This is used by
+ * {@link AuthManager} implementations, e.g. to set authentication headers on the request.
+ */
+public interface HttpRequestFacade {
+
+  /** Returns the full request URI of this request. */
+  URI getUri();
+
+  /** Returns the HTTP method of this request. */
+  String getMethod();
+
+  /** Returns the request body. Currently only String bodies are supported. */
+  Object getBody();
+
+  /** Returns all the headers of this request. */
+  Map<String, List<String>> getHeaders();
+
+  /** Returns all the headers with given name. */
+  List<String> getHeaders(String name);
+
+  /** Checks if a certain header is present in this request. */
+  default boolean containsHeader(String name) {
+    return getHeaders().containsKey(name);
+  }
+
+  /**
+   * Adds a header to this request. Existing headers with the same name will not be overwritten.
+   *
+   * @param name the name of the header.
+   * @param value the value of the header.
+   */
+  void addHeader(String name, String value);
+
+  /**
+   * Sets a header to this request. Existing headers with the same name will be overwritten.
+   *
+   * @param name the name of the header.
+   * @param value the value of the header.
+   */
+  void setHeader(String name, String value);
```

Comment:
If this replaces the header, I think it should be `replaceHeader` to be more clear.

Reply:
Wouldn't `replaceHeader` imply that if the header isn't set, then nothing should happen? Whereas the usual behavior for a setter method is to either set the value if it doesn't exist, or replace the existing one.